specs/data_structures.md

```diff
@@ -105,16 +105,17 @@ Blocks are the top-level data structure of the LazyLedger blockchain.
-| `lastCommitRoot`    | [HashDigest](#hashdigest) | Previous block's Tendermint commit root.                                     |
-| `consensusRoot`     | [HashDigest](#hashdigest) | Merkle root of [consensus parameters](#consensus-parameters) for this block. |
-| `stateCommitment`   | [HashDigest](#hashdigest) | The [state root](#state) after this block's transactions are applied.        |
-| `availableDataRoot` | [HashDigest](#hashdigest) | Root of [commitments to erasure-coded data](#availabledataheader).           |
-| `proposerAddress`   | [Address](#address)       | Address of this block's proposer.                                            |
+| name                              | type                      | description                                                                                                                                                                           |
+| --------------------------------- | ------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `height`                          | [Height](#type-aliases)   | Block height. The genesis block is at height `1`.                                                                                                                                     |
+| `timestamp`                       | [Timestamp](#timestamp)   | Timestamp of this block.                                                                                                                                                              |
+| `lastBlockID`                     | [BlockID](#blockid)       | Previous block's ID.                                                                                                                                                                  |
+| `lastCommitRoot`                  | [HashDigest](#hashdigest) | Previous block's Tendermint commit root.                                                                                                                                              |
+| `consensusRoot`                   | [HashDigest](#hashdigest) | Merkle root of [consensus parameters](#consensus-parameters) for this block.                                                                                                          |
+| `stateCommitment`                 | [HashDigest](#hashdigest) | The [state root](#state) after this block's transactions are applied.                                                                                                                 |
+| `availableDataOriginalSquareSize` | `uint64`                  | The number of rows/columns of the original data [shares](data_structures.md#share) in [square layout](data_structures.md#arranging-available-data-into-shares). Must be a power of 2. |
```

Comment:
~~Must be a power of 2.~~
Must be a square number (?)

Reply:
It's the number of rows/columns, not rows times columns. (Philosophically, you can define a rectangle as the length of two of its sides, and a square as the length of one of its sides since a square is a special rectangle).

Technically there's no absolutely hard reason we would need to restrict the size of the original square to a power of 2, but it'll make our life easier and there's not any particularly meaningful downside that I can see.

Reply:
In what way does a power of 2 make our life easier?

Reply:
If the number of shares is a power of 2, then we have the guarantee that the entire left subtree of the NMT is original data and the entire right subtree is parity data.

Reply:
While I see the beauty of this is it really worth blowing up the square? Are there other reasons why this is desireable?

Reply:
It also gives us the nice property that we can compute the structures of the paths of subtrees of different sizes and starting offsets trivially, which is needed for the non-interactive default message layout.

Reply:
An alternative would be to define a wrapper function for verifying Merkle proofs via the data root that offsets the index based on the size of each row and column, see section 5.2 in https://arxiv.org/pdf/1809.09044.pdf and the footnote.

Reply:
Ah, so here's the issue with not using a power of 2: consider the case of a max square size of 8x8. A user wants to include a message 8 shares long, and includes a signature in their transaction for a single Merkle root with 8 leaves. Now if the block producer makes a 7x7 block, the transaction won't be valid for that block, since that message would need two rows.

If square sizes are always a power of 2, then transaction senders can non-interactively include a logarithmic number of signatures with their transaction, for different square sizes. Note that the signatures for square sizes > the number of message shares will be duplicates (and can thus be deduplicated).

Reply:
> then transaction senders can non-interactively include a logarithmic number of signatures with their transaction,

I'm not sure I understand how this would work. I think this needs further clarification. Let's say user want to submit Message `(myOruNID, myOruBlob)` to LL. 
What steps does a user need to do before he can submit that message and a Tx that pays for that message? What needs to be generated (the Tx obviously) and what are the things that need to be signed?

Reply:
Also, I think we need to be clearer on how does the block producer decide on the number `k`/`availableDataOriginalSquareSize `.

Reply:
Scratch [what I said above](https://github.com/lazyledger/lazyledger-specs/pull/79#discussion_r502198374). You should be able to still sign over a logarithmic number of different share layouts regardless of the specific current square size. That being said, the formulas for the NMT currently assume that the square is a power of 2, so if we want to allow more flexibility I can look into it in a future PR that would have to change those too.